bundles/org.openhab.core/src/main/java/org/openhab/core/net/NetUtil.java

```diff
@@ -531,19 +530,20 @@ private void pollAndNotifyNetworkInterfaceAddress() {
 
         // Look for added addresses to notify
         List<CidrAddress> added = newInterfaceAddresses.stream()
-                .filter(newInterfaceAddr -> !lastKnownInterfaceAddresses.contains(newInterfaceAddr))
-                .collect(Collectors.toList());
+                .filter(newInterfaceAddr -> !lastKnownInterfaceAddresses.contains(newInterfaceAddr)).toList();
 
         // Look for removed addresses to notify
         List<CidrAddress> removed = lastKnownInterfaceAddresses.stream()
-                .filter(lastKnownInterfaceAddr -> !newInterfaceAddresses.contains(lastKnownInterfaceAddr))
-                .collect(Collectors.toList());
+                .filter(lastKnownInterfaceAddr -> !newInterfaceAddresses.contains(lastKnownInterfaceAddr)).toList();
 
         lastKnownInterfaceAddresses = newInterfaceAddresses;
 
         if (!added.isEmpty() || !removed.isEmpty()) {
-            LOGGER.debug("added {} network interfaces: {}", added.size(), Arrays.deepToString(added.toArray()));
-            LOGGER.debug("removed {} network interfaces: {}", removed.size(), Arrays.deepToString(removed.toArray()));
+            if (LOGGER.isDebugEnabled()) {
```

Comment:
This can be added to the statement above

Reply:
Which statement exactly? It always has to call `notifyListeners(added, removed);` below regardless of the log level. I only wanted to prevent the relatively expensive debug logging string operations.

Reply:
I didn't see that.